### PR TITLE
Review: Knowledge Base Page for TNS could not resolve connect error

### PR DESCRIPTION
**Pull request by rnahar25**

### CODE REVIEW

File: knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# "ORA-12154: TNS:could not resolve the connect identifier specified" Error
+
+## This error message commonly shows up for the following 2 reasons:
+1. Incorrect wallet configuration.
+   1. Follow [instructions here](https://github.com/oracle/oci-data-science-ai-samples/blob/master/environment_examples/database/autonomous_database.ipynb), if you haven't already, to download the wallet and unzip it locally.
+   2. Next, within `~/.database` directory there should be a folder with your database name. Within that will be the unzipped wallet files with some modifications made by `ads` code.  Here are two things to confirm are correct within the wallet files:
+      1. Check that `sqlnet.ora` file has the wallet location referenced. 
+      For example sqlnet.ora will have a part like `(METHOD_DATA = (DIRECTORY="/Users/rnahar/.database/Wallet_ADW")))`
```
Comment (johnpeach) [inline knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md:8]: Change "/Users/rnahar/.database/Wallet_ADW" to "/home/datascience/.database/MyDatabase"

File: knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# "ORA-12154: TNS:could not resolve the connect identifier specified" Error
+
+## This error message commonly shows up for the following 2 reasons:
+1. Incorrect wallet configuration.
+   1. Follow [instructions here](https://github.com/oracle/oci-data-science-ai-samples/blob/master/environment_examples/database/autonomous_database.ipynb), if you haven't already, to download the wallet and unzip it locally.
+   2. Next, within `~/.database` directory there should be a folder with your database name. Within that will be the unzipped wallet files with some modifications made by `ads` code.  Here are two things to confirm are correct within the wallet files:
```
Comment (johnpeach) [inline knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md:6]: "Within that will be the unzipped wallet files with some modifications made by `ads` code" Change to "Within your database name directory, your database configuration files."

Comment (johnpeach) [inline knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md:6]: Change "Here are two things to confirm are correct within the wallet files" to "There are several things to confirm:"

File: knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      1. Check that `sqlnet.ora` file has the wallet location referenced. 
+      For example sqlnet.ora will have a part like `(METHOD_DATA = (DIRECTORY="/Users/rnahar/.database/Wallet_ADW")))`
+      2. Check the connection identifier is found in the `tns_names.ora`. 
+      For example there should be a section like `dbtest123_low = (description= (retry_count=20)(retry_delay=3)(address=(protocol=tcps)(port=1522)(host=...`
+      3. Lastly check connection with sql*plus. This will confirm username/password and wallet are valid. Template for command:
+      `sqlplus <user_name>/<password>@<protocol>://<host>:<port>/<service_name>?wallet_location="<unziped_wallet_location>"`
```
Comment (johnpeach) [inline knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md:12]: You should define what each of these are and if possible give examples for things like the typical port number and that the unziped_wallet_location is the directory that was discussed above.

File: knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+2. Incorrect connection identifier (specified as `database_name` in credentials dictionary in `autonomous_database.ipynb` notebook).
+   - If this is ADW then when updating credentials, for database name we want to be using the [connection identifier](https://docs.oracle.com/cd/E11882_01/network.112/e41945/glossary.htm#BGBBGCEG) which will usually end in _high, _medium or _low. All three allow you to connect to the same database but they give different levels of number of concurrent SQL statements allowed to be run along with different CPU and IO resources available for each.
```
Comment (johnpeach) [inline knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md:14]: "If this is ADW then when updating credentials" does not make sense to me... updating credentials???

ADW should be the Autonomous Database (ADB) which is the ADW and ATP(?) databases.

File: knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+Note: This error message will not show up if you have an incorrect user name or password, in that case you will get "ORA-01017: invalid username/password; logon denied"
```
Comment (johnpeach) [inline knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md:16]: "user name" -> "username"

**Review by johnpeach — CHANGES_REQUESTED** [review]

File: knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md (hunk 2 above)
Comment (rnahar25) [reply knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md]: Updated!

Comment (rnahar25) [reply knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md]: Updated!

File: knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md (hunk 1 above)
Comment (rnahar25) [reply knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md]: Updated!

File: knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md (hunk 4 above)
Comment (rnahar25) [reply knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md:14]: In the line above I say "credentials dictionary in `autonomous_database.ipynb` notebook" so then in this line, when I say "credentials" I am referring to the same credentials dictionary. What would be better to say?

File: knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md (hunk 5 above)
Comment (rnahar25) [reply knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md]: updated!

File: knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md [hunk 6]
```diff
@@ -11,9 +11,9 @@
...
 2. Incorrect connection identifier (specified as `database_name` in credentials dictionary in `autonomous_database.ipynb` notebook).
-   - If this is ADW then when updating credentials, for database name we want to be using the [connection identifier](https://docs.oracle.com/cd/E11882_01/network.112/e41945/glossary.htm#BGBBGCEG) which will usually end in _high, _medium or _low. All three allow you to connect to the same database but they give different levels of number of concurrent SQL statements allowed to be run along with different CPU and IO resources available for each.
+   - If this is Autonomous Database (ADB) which is the ADW and ATP databases, then when updating credentials, for database name we want to be using the [connection identifier](https://docs.oracle.com/cd/E11882_01/network.112/e41945/glossary.htm#BGBBGCEG) which will usually end in _high, _medium or _low. All three allow you to connect to the same database but they give different levels of number of concurrent SQL statements allowed to be run along with different CPU and IO resources available for each.
```
Comment (johnpeach) [inline knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md:14]: Change:
If this is Autonomous Database (ADB) which is the ADW and ATP databases, then when updating credentials, for database name we want to be using the [connection identifier](https://docs.oracle.com/cd/E11882_01/network.112/e41945/glossary.htm#BGBBGCEG) which will usually end in _high, _medium or _low. All three allow you to connect to the same database but they give different levels of number of concurrent SQL statements allowed to be run along with different CPU and IO resources available for each.

To:
For most databases, the `database_name` variable is the name of the database. However, for any of the autonomous databases (ADB) you will need to use the  [connection identifier](https://docs.oracle.com/cd/E11882_01/network.112/e41945/glossary.htm#BGBBGCEG). Generally, this is the name of the database followed by `_high`, `_medium` or `_low`. However, your database may have non-standard connection identifier names.

**Review by johnpeach — CHANGES_REQUESTED** [review]

File: knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md (hunk 3 above)
Comment (rnahar25) [reply knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md]: Example added! And provided more information about the params

File: knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md (hunk 6 above)
Comment (rnahar25) [reply knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md]: Updated!

**Pull request closed without merging**